PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -3462,27 +2772,23 @@ double ILP_solver::GenerateValidSolution(const design& mydesign, const SeqPair&
         }
       }
     }
+    bool non_zero_xoffset = false, non_zero_yoffset = false;
     for (unsigned i = 0; i < mydesign.Blocks.size(); i++) {
-      bool non_zero_xoffset = false, non_zero_yoffset = false;
-      for (auto instance : mydesign.Blocks[i]) {
-        for (auto offset : instance.xoffset) {
-          if (offset != 0) {
-            non_zero_xoffset = true;
-            break;
-          }
+      for (auto& instance : mydesign.Blocks[i]) {
+        if (!instance.xoffset.empty()) {
+          non_zero_xoffset = true;
+          break;
         }
-        if (non_zero_xoffset) break;
       }
-      if (!non_zero_xoffset) roundup(Blocks[i].x, x_pitch);
-      for (auto instance : mydesign.Blocks[i]) {
-        for (auto offset : instance.yoffset) {
-          if (offset != 0) {
-            non_zero_yoffset = true;
-            break;
-          }
+      for (auto& instance : mydesign.Blocks[i]) {
+        if (!instance.yoffset.empty()) {
+          non_zero_yoffset = true;
+          break;
         }
-        if (non_zero_yoffset) break;
       }
+    }
+    for (unsigned i = 0; i < mydesign.Blocks.size(); i++) {
+      if (!non_zero_xoffset) roundup(Blocks[i].x, x_pitch);
       if (!non_zero_yoffset) roundup(Blocks[i].y, y_pitch);
     }
```

Comment:
The old code: roundup a block if it has nonzero offset.
The new code: roundup all blocks if any block has nonzero offset.

What is the reason to make this change?

Reply:
Do we need a roundup after your recent check-in?

Reply:
No, because all blocks are placed at "pitch*n+offset" now